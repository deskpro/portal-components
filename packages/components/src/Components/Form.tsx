import * as React from 'react';
import { Form as FormikForm } from 'formik';
import classNames from 'classnames';

interface IProps {
  className?: string;
  showHover?: boolean;
  children:   React.ReactNode;
}

const Form = ({
  className = '',
  showHover = true,
  ...props
}: IProps) => (
  <FormikForm
    className={classNames('dp-pc_form', className, { 'show-hover': showHover })}
    {...props}
  />);



export default Form;
