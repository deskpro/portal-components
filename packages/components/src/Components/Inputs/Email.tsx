import Field from '../Field';

class Email extends Field {
  constructor(props) {
    super(props);
    this.type = 'email';
  }
}

export default Email;
