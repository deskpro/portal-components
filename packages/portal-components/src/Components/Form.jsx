import React from 'react';
import { Form as FormikForm } from 'formik';

const Form = ({
  ...props
}) => <FormikForm {...props} />;

export default Form;
