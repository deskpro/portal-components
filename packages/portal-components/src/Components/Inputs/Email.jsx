import Input from './Input';

class Email extends Input {
  constructor(props) {
    super(props);
    this.type = 'email';
  }
}

export default Email;
