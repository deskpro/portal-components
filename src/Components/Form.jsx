import React from 'react';
import { Form as FormikForm } from 'formik';

const Form = ({
  ...props
}) => {
  return <FormikForm {...props} />
};

export default Form;