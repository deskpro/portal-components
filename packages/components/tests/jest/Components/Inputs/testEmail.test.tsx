import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { withFormik } from 'formik';
import Form from '../../../../src/Components/Form';
import Email from '../../../../src/Components/Inputs/Email';

describe('>>> Email --- Snapshot', () => {
  it('+++capturing Snapshot of Email', () => {
    const App = () => (
      <Form>
        <Email name="email" />
      </Form>
    );
    const FormikApp = withFormik({
      mapPropsToValues({
        email
      }) {
        return {
          email: email || 'test@deskpro.com'
        };
      },
      handleSubmit(values, { setSubmitting }) {
        setSubmitting(false);
      }
    })(App);
    const renderedValue = renderer.create(<FormikApp />).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
});
