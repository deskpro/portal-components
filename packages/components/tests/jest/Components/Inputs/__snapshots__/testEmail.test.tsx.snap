// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>> Email --- Snapshot +++capturing Snapshot of Email 1`] = `
<form
  action="#"
  className="dp-pc_form show-hover"
  onReset={[Function]}
  onSubmit={[Function]}
>
  <div
    className="dp-pc_field"
  >
    <label
      className="dp-pc_label"
      htmlFor="field_1"
    >
      
    </label>
    <input
      className="dp-pc_input"
      id="field_1"
      name="email"
      onBlur={[Function]}
      onChange={[Function]}
      required={false}
      type="email"
      value="test@deskpro.com"
    />
  </div>
</form>
`;
